### PR TITLE
compile terraform-boot API

### LINKED ISSUE #757: compile terraform-boot API
- [ ] add a new package under "org.eclipse.xpanse.modules.deployment.deployers.terraform" with name "terraformboot"
- [ ] Copy the openapi json file from terraform-boot API under resources folder of "deployment" folder. 
- [ ] Add the openapi-generator maven plugin  which reads this JSON and generates the API client and model. (similar to what we do on  UI to generate the client and models for xpanse API)
- [ ] the maven plugin must be only executed on demand. It must be disabled by default. 
- [ ] Document how to run the plugin and generate the client again.
- [ ] Commit the API JSON, data models and client to GitHub

### PR DESCRIPTION
**Pull request by jinyangyang222**
fixes #757

### CODE REVIEW

File: modules/deployment/pom.xml [hunk 1]
```diff
@@ -8,6 +8,10 @@
         <version>1.0.0-SNAPSHOT</version>
         <relativePath>../pom.xml</relativePath>
     </parent>
+    <properties>
+        <openapiGenerator.skip>true</openapiGenerator.skip>
```
Comment (swaroopar) [inline modules/deployment/pom.xml:12]: @jinyangyang222 can you please rename this to `generate.terraform.boot.client` and update the readme also accordingly.

File: README.md [hunk 2]
```diff
@@ -33,6 +33,13 @@ Details can be found on the project website [here](https://eclipse-xpanse.github
 
 Details can be found on the project website [here](https://eclipse-xpanse.github.io/xpanse-website/docs/database).
 
+## Deployment
```
Comment (swaroopar) [inline README.md:36]: headline must be - Generate terraform-boot client code

File: README.md [hunk 3]
```diff
@@ -33,6 +33,13 @@ Details can be found on the project website [here](https://eclipse-xpanse.github
...
+
```
Comment (swaroopar) [inline README.md:37]: Also document the step about how to get the yaml file and with which name it must be copied to which location.

**Review by swaroopar — CHANGES_REQUESTED** [review]
@jinyangyang222 looks great. Just minor comments added.

File: README.md (hunk 3 above)
Comment (jinyangyang222) [reply README.md:37]: start terraform-boot to access“ http://localhost:9090/v3/api-docs" get json file and convert it to yaml format

Comment (jinyangyang222) [reply README.md:37]: json file can also be used directly without converting to yaml.

File: modules/deployment/pom.xml (hunk 1 above)
Comment (jinyangyang222) [reply modules/deployment/pom.xml]: fixed

File: README.md (hunk 2 above)
Comment (jinyangyang222) [reply README.md]: fixed

**Review by swaroopar — APPROVED** [review]
LGTM

**Pull request merged**